### PR TITLE
Diet benchmarks

### PR DESCRIPTION
**Pull request by anicolaspp**
I am completely new to JMH so please, review the small benchmarking code

### CODE REVIEW

File: tests/src/test/scala/DietSpec.scala [hunk 1]
```diff
@@ -115,6 +115,13 @@ class DietTest extends FlatSpec with Matchers {
   import dogs.Predef._
   import Diet._
 
+  implicit object EnumInt extends Enum[Int] {
```
Comment (stew) [inline tests/src/test/scala/DietSpec.scala]: we don't have to define this in the test, we have it in main:

https://github.com/stew/dogs/blob/master/core/src/main/scala/Enum.scala#L28

File: tests/src/test/scala/DietSpec.scala [hunk 2]
```diff
@@ -239,7 +238,13 @@ class DietTest extends FlatSpec with Matchers {
 }
 
 class DietTestJoin extends FlatSpec with Matchers {
-  import Diet._
+
+  implicit object EnumInt extends Enum[Int] {
```
Comment (stew) [inline tests/src/test/scala/DietSpec.scala]: we already have this in main

**Comment by stew** [comment]
this needs to be rebased against master

**Comment by anicolaspp** [comment]
I pull requested my master after rebase it. I think we can close this one now.

**Pull request closed without merging**